envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/EnvoySnapshotFactory.kt

```diff
@@ -43,9 +45,12 @@ internal class EnvoySnapshotFactory(
                 idleTimeout = Durations.fromMillis(properties.egress.commonHttp.idleTimeout.toMillis()),
                 requestTimeout = Durations.fromMillis(properties.egress.commonHttp.requestTimeout.toMillis())
             )
-        )
+        ),
+    private val meterRegistry: MeterRegistry
 ) {
     fun newSnapshot(servicesStates: List<LocalityAwareServicesState>, ads: Boolean): Snapshot {
+        val sample = Timer.start(meterRegistry)
```

Comment:
How about `@Timed` annotation using aspects? It seems to be less code interfering